src/neural/network_check.cc

```diff
@@ -77,48 +99,114 @@ class CheckComputation : public NetworkComputation {
 
     bool policyAlmostEqual = true;
     for (int i = 0; i < size && policyAlmostEqual; i++) {
-      for (int j = 0; j < NUM_OUTPUT_POLICY; j++) {
+      for (int j = 0; j < kNumOutputPolicies; j++) {
         float v1 = refComp_->GetPVal(i, j);
         float v2 = checkComp_->GetPVal(i, j);
         policyAlmostEqual &= IsAlmostEqual(v1, v2);
       }
     }
 
-    // We could also print on standard output using the uci info string
-    // example:
-    //
-    // info string Check passed
-    // info string ***** error check failed
-    //
-
     if (valueAlmostEqual && policyAlmostEqual) {
-      std::cerr << "Check passed for a batch of " << size << std::endl;
+      fprintf(stderr, "Check passed for a batch of %d.\n", size);
```

Comment:
We use C++ streams in this project, it's better to use them too for consistency.

Reply:
All the backends use stdio. Check yourself.
Histogram.cc relies heavily on stdio and all the related varargs stuff.